### ISSUE TITLE
[Feature] Quota settings should be validated before being applied

### ISSUE THREAD

**Issue by shan916** [Axway-API-Management-Plus/apimanager-swagger-promote#109]
**Swagger-Promote Version**
1.6.0

**API-Manager and Service-Pack Version**
7.5.3 SP7

**Expected behavior**
The tool should validate the quota settings before applying them. The [`systemQuota` Wiki example](https://github.com/Axway-API-Management-Plus/apimanager-swagger-promote/wiki/2.3-Configure-API-Quotas) example incorrectly shows **minutes** instead of **minute**.

**Actual behavior**
The tool reports that the quota was successfully applied with `period`=minutes but API Manager shows the incorrect configuration. 
![image](https://user-images.githubusercontent.com/4755022/61165220-6967ab80-a4d2-11e9-8c77-b79f7a50a043.png)

Invoking the API causes a 500 at runtime.
```
Caused by: java.lang.IllegalArgumentException: No enum constant com.vordel.apiportal.runtime.quota.constraint.TimeWindowUnit.minutes
at java.lang.Enum.valueOf(Enum.java:238)
```

**Test case sample**
> Create an API config as shown in the Wiki with period set to **minutes** instead of **minute** and run the tool. Check the configuration in API Manager.

**Comment by cwiechmann** [comment]
@shan916, good catch. Have just updated the wiki (that was easy). I'm now off for the next three weeks, after that will work on this:
- create a test-case: InvalideQuotaConfig
- implement a validation

**Comment by cwiechmann** [comment]
Added a validation for the QuotaConfiguration plus an IntegrationTest validates that this is working as expected. If the Quota-Config is invalid a new ReturnCode 71 is used on exit. The API will not be replicated at all.
Also added a validation for the configured Inbound- and Outbound-Security-Types that only valid types are given.

**Comment by cwiechmann** [comment]
Has been fixed/enhanced in release 1.6.2.
